src/pyEQL/solution.py

```diff
@@ -37,6 +37,87 @@
 UNKNOWN_OXI_STATE = "unk"
 K_W = 1e-14  # ion product of water at 25 degC
 
+from pyEQL import ureg
+
+# Extend the UnitRegistry with custom units
+ureg.define('ppb = 1e-9 * mole / mole = parts_per_billion')
+ureg.define('ppm = 1e-6 * mole / mole = parts_per_million')
+ureg.define('% = 0.01 * mole / mole = percent')
+
+
+def preprocess_units(amount: str) -> str:
+    """
+    Preprocess the unit string to handle custom units like 'ppb' and '%'.
+    Args:
+        amount (str): The input amount with units (e.g., '1000 ppb', '0.1%').
+    Returns:
+        str: A modified amount string compatible with Pint.
+    """
+    if "ppb" in amount:
+        return amount.replace("ppb", "microgram/liter")
+    elif "%" in amount:
+        # Convert '%' to a dimensionless fraction
+        value, unit = amount.split()
+        return f"{float(value) / 100} dimensionless"
+    return amount  # Return unchanged if no custom units
```

Comment:
I like the idea of a dedicated `preprocess_units` method (see comments below); however there should not be two versions of it. Currently there is this one and another one in the middle of the `__init__` method. See other comment below.

Reply:
I realized that what you proposed actually parallels what is already done in `get_amount`. There is a separate utility method called [`interpret_units`](https://github.com/KingsburyLab/pyEQL/blob/5ef36e5f2591cb3685b5f8981bcf5038ecd28333/src/pyEQL/utils.py#L22) that does exactly what you're trying to do here. I think addressing this bug might be as simple as replacing your `preprocess_units` in `add_solute` with the existing `interpret_units`